### PR TITLE
openaire: added award title and acronym to serialization.

### PR DESCRIPTION
**Pull request by alejandromumo**
closes https://github.com/zenodo/ops/issues/313

### CODE REVIEW

File: site/zenodo_rdm/openaire/serializers/schema.py [hunk 1]
```diff
@@ -196,10 +196,15 @@ def _reverse_funder_acronym(funder_ror):
...
                 if funder_acronym and award_program and award_number:
-                    links.append(
-                        f"info:eu-repo/grantAgreement/{funder_acronym}/{award_program}/{award_number}"
-                    )
+                    b_link = f"info:eu-repo/grantAgreement/{funder_acronym}/{award_program}/{award_number}"
+                    if award_title:
+                        b_link += f"/{award_title}"
+                    if award_acronym:
+                        b_link += f"/{award_acronym}"
+                    links.append(b_link)
```
Comment (slint) [inline site/zenodo_rdm/openaire/serializers/schema.py:207]: major: if `award_title` or `award_acronym` is missing, then we still include them in the final string as empty values. i.e. if both were empty we would just have `info:eu-repo/grantAgreement/EC/H2020/12345//`. See for reference the [logic in legacy](https://github.com/zenodo/zenodo/blob/482ee72ad501cbbd7f8ce8df9b393c130d1970f7/zenodo/modules/openaire/schema.py#L132-L135).

File: site/zenodo_rdm/openaire/serializers/schema.py [hunk 2]
```diff
@@ -196,10 +196,15 @@ def _reverse_funder_acronym(funder_ror):
...
                 award_number = award.get("number", "")
+                award_title = award.get("title", {}).get("en")
+                award_acronym = award.get("acronym", "")
                 if funder_acronym and award_program and award_number:
-                    links.append(
-                        f"info:eu-repo/grantAgreement/{funder_acronym}/{award_program}/{award_number}"
-                    )
+                    b_link = f"info:eu-repo/grantAgreement/{funder_acronym}/{award_program}/{award_number}"
+                    if award_title:
+                        b_link += f"/{award_title}"
```
Comment (slint) [inline site/zenodo_rdm/openaire/serializers/schema.py:204]: major: title and acronyms might include slashes `/`. In legacy we [URL-encoded them](https://github.com/zenodo/zenodo/blob/482ee72ad501cbbd7f8ce8df9b393c130d1970f7/zenodo/modules/openaire/schema.py#L134). I would apply the same transformation all the fields (i.e. `funder_acronym`, `award_program`, `award_number`, `award_title`, `award_acronym`)

**Review by slint — COMMENTED** [review]
Two changes to comply with what we had in legacy.

File: site/zenodo_rdm/openaire/serializers/schema.py (hunk 2 above)
Comment (alejandromumo) [reply site/zenodo_rdm/openaire/serializers/schema.py]: I encoded them using [urllib.parse.quote_plus](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote_plus) for complete URL encoding. I.e. spaces were not being URL encoded in legacy.

**Comment by alejandromumo** [comment]
Example of a record:

```python
{'originalId': '10.5281/zenodo.202',
 'title': 'esrqwewqqeqw',
 'url': 'https://127.0.0.1:5000/records/202',
 'authors': ['asdqwewqewqas'],
 'licenseCode': 'OPEN',
 'publisher': 'Zenodo',
 'collectedFromId': 'opendoar____::2659',
 'hostedById': 'opendoar____::2659',
 'linksToProjects': ['info:eu-repo/grantAgreement/NIH/NATIONAL_HEART%2C_LUNG%2C_AND_BLOOD_INSTITUTE/5P50HL060293-05IMPACT+OF+AIRWAY+INFECTION+ON+ADENOVIRAL+GENE+TRANSFER',
  'info:eu-repo/grantAgreement/NSF/ENG%2FOAD/7620280Uses+of+Modern+System+Theory+in+Water+Resources'],
 'pids': [{'type': 'doi', 'value': '10.5281/zenodo.202'},
  {'type': 'oai', 'value': 'oai:zenodo.org:202'}],
 'type': 'dataset',
 'resourceType': '0027'}
```

**Pull request merged**